doc/driver.mld

```diff
@@ -816,99 +813,104 @@ This last block analyze the running times so that they can be submitted to
-      ]
+        ("name", `String ("total-" ^ cmd));
+        ("value", `Int count);
+        ( "description",
+          `String ("Number of time 'odoc " ^ cmd ^ "' has run.") );
+      ];
+    `Assoc
+      [
+        ("name", `String ("time-" ^ cmd));
+        ( "value",
+          `Assoc
+            [
+              ("min", `Float min); ("max", `Float max); ("avg", `Float avg);
+            ] );
+        ("units", `String "s");
+        ("description", `String ("Time taken by 'odoc " ^ cmd ^ "'"));
+        ("trend", `String "lower-is-better");
+      ];
+  ]
 
 (** Analyze the size of files produced by a command. *)
 let compute_produced_cmd cmd =
   let output_file_size c =
     match c.output_file with
     | Some f -> (
         match Bos.OS.Path.stat f with
-        | Ok st -> Some st.Unix.st_size
+        | Ok st -> Some (float st.Unix.st_size)
         | Error _ -> None)
     | None -> None
   in
-  let rec compute min_ max_ total count = function
-    | [] -> (min_, max_, int_of_float (total /. float count), count)
-    | size :: tl ->
-        compute (min min_ size) (max max_ size) (total +. float size) (count + 1) tl
-  in
-  match List.filter_map output_file_size (filter_commands cmd) with
-  | [] -> failwith ("No commands run for " ^ cmd)
-  | size0 :: tl ->
-      let min, max, avg, count = compute size0 size0 (float size0) 1 tl in
+  let sizes = List.filter_map output_file_size (filter_commands cmd) in
+  let min, max, avg, count = compute_min_max_avg sizes in
+  let min = int_of_float min in
+  let max = int_of_float max in
+  let avg = int_of_float avg in
+  [
+    `Assoc
       [
-        `Assoc
-          [
-            ("name", `String ("produced-total-" ^ cmd));
-            ("value", `Int count);
-            ( "description",
-              `String ("Number of file produced by 'odoc " ^ cmd ^ "'") );
-          ];
-        `Assoc
-          [
-            ("name", `String ("produced-size-" ^ cmd));
-            ( "value",
-              `Assoc
-                [
-                  ("min", `Int min); ("max", `Int max); ("avg", `Int avg);
-                ] );
-            ("units", `String "b");
-            ( "description",
-              `String ("Size of file produced by 'odoc " ^ cmd ^ "'") );
-            ("trend", `String "lower-is-better");
-          ];
-      ]
+        ("name", `String ("produced-total-" ^ cmd));
+        ("value", `Int count);
+        ( "description",
+          `String ("Number of file produced by 'odoc " ^ cmd ^ "'") );
+      ];
+    `Assoc
+      [
+        ("name", `String ("produced-size-" ^ cmd));
+        ( "value",
+          `Assoc
+            [
+              ("min", `Int min); ("max", `Int max); ("avg", `Int avg);
+            ] );
+        ("units", `String "b");
+        ( "description",
+          `String ("Size of file produced by 'odoc " ^ cmd ^ "'") );
+        ("trend", `String "lower-is-better");
+      ];
+  ]
 
 (** Analyze the size of files produced by a command. *)
```

Comment:
This comment looks like is the result of a (previous) copy-paste!.

Reply:
Nice catch!